rotkehlchen/chain/evm/decoding/curve_lend/decoder.py

```diff
@@ -617,14 +632,110 @@ def _decode_vault_controller_events(self, context: DecoderContext) -> DecodingOu
+                event.notes = f'Deposit {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} into {gauge_asset.symbol}'  # noqa: E501
+
+                paired_events_data = ([event], True)
+                from_event_type = HistoryEventType.RECEIVE
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.RECEIVE
+                to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                to_notes = f'Receive {amount} {gauge_asset.symbol} after depositing in curve lending vault gauge'  # noqa: E501
+                break
+
+            if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == amount and
+                    event.address == context.tx_log.address
+            ):
+                event.counterparty = CPT_CURVE
+                event.product = EvmProduct.GAUGE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Withdraw {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from {gauge_asset.symbol}'  # noqa: E501
+
+                paired_events_data = ([event], False)
+                from_event_type = HistoryEventType.SPEND
+                from_event_subtype = HistoryEventSubType.NONE
+                to_event_type = HistoryEventType.SPEND
+                to_event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                to_notes = f'Return {amount} {gauge_asset.symbol} after withdrawing from curve lending vault gauge'  # noqa: E501
+                break
+        else:
+            log.error(f'Failed to find deposit/withdraw event for curve lending vault gauge for {context.transaction}')  # noqa: E501
+
+        return DecodingOutput(action_items=[] if paired_events_data is None else [
+            ActionItem(
+                action='transform',
+                from_event_type=from_event_type,
+                from_event_subtype=from_event_subtype,
+                asset=gauge_asset,
+                amount=amount,
+                to_counterparty=CPT_CURVE,
+                to_notes=to_notes,
+                paired_events_data=paired_events_data,
+                to_event_type=to_event_type,
+                to_event_subtype=to_event_subtype,
+            ),
+        ])
+
+    def _maybe_enrich_curve_lending_gauge_rewards(self, context: EnricherContext) -> TransferEnrichmentOutput:  # noqa: E501
+        """ This enriches rewards paid to users from the vault's gauge contract.
+        It is always a transfer of the reward tokens of the gauge without any other log emitted.
+
+        May raise:
+        - UnknownAsset
+        - WrongAssetType
+        """
+        source_address = bytes_to_address(context.tx_log.topics[1])
```

Comment:
Since this is an enricher the very first check in the if should be the address check. Quit as fast as possible.